x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/execution/search/extractor/AbstractFieldHitExtractor.java

```diff
@@ -81,12 +83,65 @@ public Object extract(SearchHit hit) {
         Object value = null;
         DocumentField field = null;
         if (hitName != null) {
-            // a nested field value is grouped under the nested parent name (ie dep.dep_name lives under "dep":[{dep_name:value}])
-            field = hit.field(hitName);
+            value = extractNestedField(hit);
         } else {
             field = hit.field(fieldName);
+            if (field != null) {
+                value = unwrapFieldsMultiValue(field.getValues());
+            }
```

Comment:
This should be common for both branches (needs changes to `extractNestedField`) to make it more consistent. Something like:
```
field = hitName != null  ? extractNestedField(hit) : hit.field(fieldName);
if (field != null) {
    value = unwrapFieldsMultiValue(fields.getValue());
}
```

Reply:
`extractNestedField` cannot consistently return a `DocumentField` instance so that we can call allways `field.getValues()` in `extract(SearchHit)` method. The data structure inside a `DocumentField` is more complex in the case of nested fields and it requires further exploration before obtaining an equivalent of `field.getValues()`  (see https://github.com/elastic/elasticsearch/pull/69368/files#diff-099b1feb3ed943ecc359be919a1dd67423938b9b8252e43c9ddd0a72027c1c87R120-R143).